src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyFacet.java

```diff
@@ -120,40 +153,37 @@ protected Content store(final Context context, final Content content) throws IOE
     TokenMatcher.State matcherState = matcherState(context);
     String assetPath = buildAssetPath(context);
 
+    ConanCoords conanCoords = getCoords(context);
     if (assetKind.equals(CONAN_SRC)) {
-      return putPackage(assetPath, content, group(matcherState), project(matcherState), version(matcherState));
+      return putPackage(assetPath, content, conanCoords);
     }
-    return putMetadata(assetPath, content, assetKind, group(matcherState), project(matcherState), version(matcherState));
+    return putMetadata(assetPath, content, assetKind, conanCoords);
   }
 
   private Content putPackage(final String assetPath,
                              final Content content,
-                             final String group,
-                             final String project,
-                             final String version) throws IOException {
+                             final ConanCoords coords) throws IOException {
     StorageFacet storageFacet = facet(StorageFacet.class);
     try (TempBlob tempBlob = storageFacet.createTempBlob(content.openInputStream(), HASH_ALGORITHMS)) {
-      return doPutPackage(assetPath, project, version, tempBlob, content, group);
+      return doPutPackage(assetPath, tempBlob, content, coords);
     }
   }
 
   private Content putMetadata(final String assetPath,
                               final Content content,
                               final AssetKind assetKind,
-                              final String group,
-                              final String project,
-                              final String version)
+                              final ConanCoords coords)
       throws IOException
   {
     StorageFacet storageFacet = facet(StorageFacet.class);
     try (TempBlob tempBlob = storageFacet.createTempBlob(content.openInputStream(), HASH_ALGORITHMS)) {
       AttributesMap attributesMap;
       switch (assetKind) {
         case DOWNLOAD_URL:
-          String indexAssetName = getProjectIndexName(group, project, version);
-          try(TempBlob updatedBlob = absoluteUrlRemover.updateAbsoluteUrls(tempBlob, getRepository(), indexAssetName)) {
+          String indexAssetName = getProjectIndexName(coords);
+          try (TempBlob updatedBlob = conanUrlIndexer.updateAbsoluteUrls(tempBlob, getRepository(), indexAssetName)) {
             attributesMap = ConanManifest.parse(updatedBlob);
-            return doSaveMetadata(assetPath, tempBlob, content, assetKind, attributesMap, project, version, group);
+            return doSaveMetadata(assetPath, updatedBlob, content, assetKind, attributesMap, coords);
```

Comment:
QUESTION: Was this a bug previously? Just noticed the switch to `updatedBlob` but saw you were previously using `tempBlob`?

Reply:
It was a bug :)